nodebuilder/header/module.go

```diff
@@ -19,7 +19,7 @@ import (
 
 var log = logging.Logger("module/header")
 
-func ConstructModule(tp node.Type, cfg *Config) fx.Option {
+func ConstructModule(tp node.Type, network modp2p.Network, cfg *Config) fx.Option {
```

Comment:
The network is provided to DI; we don't need to pass it, like you did for `newSubscriber`